src/object/srv_obj_migrate.c

```diff
@@ -3799,26 +3614,131 @@ migrate_try_obj_insert(struct migrate_pool_tls *tls, uuid_t co_uuid,
 	return rc;
 }
 
+struct ds_pool_migrate_arg {
+	uuid_t          pool_uuid;
+	struct ds_pool *pool;
+	uint32_t        rebuild_ver;
+	uint32_t        generation;
+	daos_prop_t    *prop;
+	int             tgt_id;
+	uint8_t         tgt_status;
+	uint32_t        tgt_in_ver;
+	int             rebuilding_count;
+	bool            no_iv;
+};
+
+static int
+ds_migrate_end_ult(void *arg)
+{
+	struct ds_pool_migrate_arg *pool_arg = (struct ds_pool_migrate_arg *)arg;
+
+	if (pool_arg->pool) {
+		pool_arg->pool->sp_rebuilding += pool_arg->rebuilding_count;
```

Comment:
sp_rebuilding's operation (++, +=, -- etc) is not atomic and looks unsafe to me, as all VOS tgt XS may do the operation.
original master code also with this problem, this PR added a few more operations maybe easier to trigger.